src/webots/core/WbSysInfo.cpp

```diff
@@ -90,6 +91,13 @@ const QString &WbSysInfo::openGLVersion() {
     openGLVersion = (const char *)glGetString(GL_VERSION);
   return openGLVersion;
 }
+const QString &WbSysInfo::defaultALDevices() {
+  const ALCchar *defaultALDevices = alcGetString(NULL, ALC_DEFAULT_DEVICE_SPECIFIER);
+  ALCdevice *gDefaultDevice = alcOpenDevice(defaultALDevices);
+  static QString defaultDeviceName = (const char *)gDefaultDevice;
+  alcCloseDevice(gDefaultDevice);
+  return defaultDeviceName;
+}
```

Comment:
That seems better:
1. It saves a few warnings when trying to close a NULL device.
2. The value for the device name was wrong.
3. It performs the evaluation only once, not at every call of the function.

```suggestion
const QString &WbSysInfo::defaultALDevices() {
  static QString defaultDeviceName = "?";
  if (defaultDeviceName != "?")
    return defaultDeviceName;
  defaultDeviceName = "";
  const ALCchar *defaultALDevice = alcGetString(NULL, ALC_DEFAULT_DEVICE_SPECIFIER);
  if (defaultALDevice) {
    ALCdevice *gDefaultDevice = alcOpenDevice(defaultALDevice);
    if (gDefaultDevice) {
      defaultDeviceName = (const char *)defaultALDevice ;
      alcCloseDevice(gDefaultDevice);
    }
  }
  return defaultDeviceName;
}
```